File: 905.go
func sortArrayByParity(nums []int) []int {
	end := len(nums) - 1
	for i := 0; i < end; i++ {
		if nums[i]%2 == 1 {
			nums[i], nums[end] = nums[end], nums[i]
			end--
			i--
		}
	}

	return nums
}

File: 905.rs
impl Solution {
    pub fn sort_array_by_parity(mut nums: Vec<i32>) -> Vec<i32> {
        let mut idx = 0;
        let mut end = nums.len() - 1;
        while idx < end {
            if nums[idx] % 2 == 1 {
                nums.swap(idx, end);
                end -= 1;
            } else {
                idx += 1;
            }
        }
        
        nums
    }
}
